packages/flutter_tools/lib/src/migrate/migrate_update_locks.dart

```diff
@@ -0,0 +1,122 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/terminal.dart';
+import '../project.dart';
+import 'migrate_utils.dart';
+
+/// Checks if the project uses pubspec dependency locking and prompts if
+/// the pub upgrade should be run.
+Future<void> updatePubspecDependencies(
+    FlutterProject flutterProject,
+    MigrateUtils migrateUtils,
+    Logger logger,
+    Terminal terminal) async {
+  final File pubspecFile = flutterProject.directory.childFile('pubspec.yaml');
+  if (!pubspecFile.existsSync()) {
+    return;
+  }
+  if (!pubspecFile.readAsStringSync().contains('# THIS LINE IS AUTOGENERATED')) {
+    return;
+  }
+  logger.printStatus('\nDart dependency locking detected in pubspec.yaml.');
+  String selection = 'y';
+  try {
+    selection = await terminal.promptForCharInput(
+      <String>['y', 'n'],
+      logger: logger,
+      prompt: 'Do you want the tool to run `flutter pub upgrade --major-versions`? (y)es, (n)o',
+      defaultChoiceIndex: 1,
+    );
+  } on StateError catch(e) {
+    logger.printError(
+      e.message,
+      indent: 0,
+    );
+  }
+  if (selection == 'y') {
+    // Runs `flutter pub upgrade --major-versions`
+    await migrateUtils.flutterPubUpgrade(flutterProject.directory.path);
+  }
+}
+
+/// Checks if gradle dependency locking is used and prompts the developer to
+/// remove and back up the gradle dependenc lockfile.
+Future<void> updateGradleDependencyLocking(
+    FlutterProject flutterProject,
+    MigrateUtils migrateUtils,
+    Logger logger,
+    Terminal terminal,
+    bool verbose,
+    FileSystem fileSystem) async {
+  final Directory androidDir = flutterProject.directory.childDirectory('android');
+  if (!androidDir.existsSync()) {
+    return;
+  }
+  final List<FileSystemEntity> androidFiles = androidDir.listSync();
+  final List<File> lockfiles = <File>[];
+  final List<String> backedUpFilePaths = <String>[];
+  for (final FileSystemEntity entity in androidFiles) {
+    if (entity is! File) {
+      continue;
+    }
+    final File file = entity.absolute;
+    // Don't re-handle backed up lockfiles.
+    if (file.path.contains('_backup_')) {
+      continue;
+    }
+    try {
+      // lockfiles generated by gradle start with this prefix.
+      if (file.readAsStringSync().startsWith('# This is a Gradle generated file for dependency locking.\n# Manual edits can break the build and are not advised.\n# This file is expected to be part of source control.')) {
```

Comment:
nit: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters